src/lazy/text/buffer.rs

```diff
@@ -3006,13 +3028,22 @@ mod tests {
             "(:foo (1 2 3))",
             "(:foo \"foo\")",
             "(:foo foo)",
+            "(:4)",
+            "(:4 1)",
+            "(:4 1 2 3)",
+            "(:4 (1 2 3))",
+            "(:4 \"foo\")",
+            "(:4 foo)",
         ],
         expect_mismatch: [
             "foo",   // No parens
             "(foo)", // No `:` after opening paren
+            "(4",    // No parens
+            "(4)",   // No `:` after opening paren
```

Comment:
Here's a couple more test cases that might fail right now because it looks like you're using the same matcher as for integer values.

```suggestion
            "(4)",   // No `:` after opening paren
            "(:0x4)", // Hexadecimal not allowed
            "(:4_000)", // underscores not allowed in number
```